### PR TITLE
LKE-7037 feat(MV): Format mv in captions

### PR DESCRIPTION
**Pull request by aminerhanemi**
formating monetary values in captions.
For that we pass the public schema from the frontend in captionsService

### CODE REVIEW

File: src/captions/captions.ts [hunk 1]
```diff
@@ -99,12 +117,25 @@ export class Captions {
...
 
+  public static getPropertyType(
+    graphSchema: GraphSchemaTypeWithAccess[],
+    propertyKey: string,
+    itemType: string
+  ): PropertyType | undefined {
+    const typeGraphSchema = graphSchema.find((schemaType) => schemaType.itemType === itemType);
+    const property = (typeGraphSchema?.properties.find(
+      (property) => property.propertyKey === propertyKey
+    ) as unknown) as GraphSchemaPropertyWithAccess;
```
Comment (aminerhanemi) [inline src/captions/captions.ts:128]: It is weird that I need a cast here, but I was not able to make it work without.
Properties is an array of GraphSchemaPropertyWithAccess so the cast shouldn't be needed or at least the cast to unknown.
```
export interface GraphSchemaTypeWithAccess extends GraphSchemaType {
    access: ItemTypeAccessRightType;
    properties: Omit<GraphSchemaProperty, 'indexedAs'>[] | GraphSchemaPropertyWithAccess[];
}
```

Comment (EricMallard) [reply src/captions/captions.ts:128]: image

image

working for me I think, could it be your IDE?

Comment (aminerhanemi) [reply src/captions/captions.ts:128]: Screen Shot 2022-09-05 at 6 05 34 PM
this is what I get

Comment (EricMallard) [reply src/captions/captions.ts:128]: actually I have a weird error but only when transpiling with 'tsc': Property 'propertyType' does not exist on type 'NonNullable<NonNullable<T>>'
was that your problem?

Comment (aminerhanemi) [reply src/captions/captions.ts:128]: Screen Shot 2022-09-05 at 6 07 38 PM

Comment (aminerhanemi) [reply src/captions/captions.ts]: PS: it fails also in the CI

Comment (aminerhanemi) [reply src/captions/captions.ts]: I think something with the Omit is not right, but not sure it is the right time to look at that. I will keep it in mind and check with backend developer after

**Review by EricMallard — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -27,12 +27,12 @@
...
-    "latest:develop": "npm install --save-exact @linkurious/rest-client@lk-develop",
-    "latest:develop-next": "npm install --save-exact @linkurious/rest-client@lk-develop-next",
+    "latest:develop-stable": "npm install --save-exact @linkurious/rest-client@lk-develop-stable @linkurious/shared@lk-develop-stable",
+    "latest:develop": "npm install --save-exact @linkurious/rest-client@lk-develop @linkurious/shared@lk-develop",
+    "latest:develop-next": "npm install --save-exact @linkurious/rest-client@lk-develop-next @linkurious/shared@lk-develop-next",
     "latest:master-maintenance": "npm install --save-exact @linkurious/rest-client@lk-master-maintenance",
-    "latest:master-stable": "npm install --save-exact @linkurious/rest-client@lk-master-stable",
-    "latest:master": "npm install --save-exact  @linkurious/rest-client@lk-master",
+    "latest:master-stable": "npm install --save-exact @linkurious/rest-client@lk-master-stable @linkurious/shared@lk-master-stable",
+    "latest:master": "npm install --save-exact  @linkurious/rest-client@lk-master @linkurious/shared@lk-master",
```
Comment (gmasclet) [inline package.json:35]: @enys fyi `@linkurious/shared` has been added as a dependency of `@linkurious/ogma-linkurious-parser` with this PR.

I've updated the `latest:*` npm scripts accordingly. I did not change the maintenance scripts has this feature is not targetting maintenance. This is probably going to bite us at some point when we promote 3.2 to stable. Feel free to open a PR with the necessary adjustments (if any) and ping me for review.

**Pull request merged**